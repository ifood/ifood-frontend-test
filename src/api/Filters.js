import axios from 'axios'

const baseURL = 'http://www.mocky.io/v2/5a25fade2e0000213aa90776'

export const getFilterAPI = () => axios.get(baseURL)
