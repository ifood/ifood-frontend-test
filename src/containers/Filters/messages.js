import { defineMessage } from 'react-intl';

const messages = {
  title: {
    id: 'filters.title',
    defaultMessage: 'Filter by:',
  },

  fields: {
    locale: {
      id: 'filters.fields.locale',
      defaultMessage: 'Locale',
    },
    country: {
      id: 'filters.fields.country',
      defaultMessage: 'Country',
    },
    timestamp: {
      id: 'filters.fields.timestamp',
      defaultMessage: 'Date',
    },
    limit: {
      id: 'filters.fields.limit',
      defaultMessage: 'Limit',
    },
    offset: {
      id: 'filters.fields.offset',
      defaultMessage: 'Offset',
    },
  },

  errors: {
    unknown: {
      id: 'filters.error.unknown',
      defaultMessage: 'An error occurred while loading the filters.',
    },
  },
};

export default defineMessage(messages);
