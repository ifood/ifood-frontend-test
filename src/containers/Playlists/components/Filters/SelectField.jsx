
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';

class SelectField extends Component {
  renderValues = () => {
    const { field } = this.props;

    return field.values.map((entry) => {
      return (
        <MenuItem key={entry.value} value={entry.value}>
          {entry.name}
        </MenuItem>
      );
    });
  }

  render() {
    const { field, onChange, value } = this.props;

    if (!field) {
      return null;
    }

    return (
      <TextField
        id={field.id}
        select
        label={field.name}
        value={value}
        fullWidth
        onChange={onChange}
        margin="normal"
      >
        {this.renderValues()}
      </TextField>
    );
  }
}

SelectField.propTypes = {
  field: PropTypes.object,
  onChange: PropTypes.func,
  value: PropTypes.string,
};

SelectField.defaultProps = {
  field: null,
  onChange: () => false,
  value: '',
};

export default SelectField;
