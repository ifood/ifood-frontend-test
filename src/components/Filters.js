/*eslint-disable*/
import React from 'react';
import useFilters from '../hooks/useFilters';
import { FiltersContainer, FilterField, Filter } from '../styles/FiltersStyles';

export default function Filters() {
  const { loading, filters } = useFilters();

  return (
    <FiltersContainer>
      {filters.map((filter) => (
        <Filter key={filter.id}>
          <FilterField label={filter.name} />
        </Filter>
      ))}
    </FiltersContainer>
  );
}
