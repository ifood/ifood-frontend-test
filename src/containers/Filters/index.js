import React, { useState, useEffect, useCallback, useMemo } from "react";
import { useSelector, useDispatch } from "react-redux";

import Loader from "components/Loader";
import useEffectUpdate from "hooks/useEffectUpdate";
import FiltersApiService from "services/filters";
import PlaylistsApiService from "services/playlists";
import { getFilterField } from "helpers/getFilterField";
import { filtersContainerData } from "constants/data/containers/Filters";

import {
  updatePlaylists,
  removePlaylists,
  removeFilteredPlaylists,
} from "store/modules/playlists/actions";

import * as S from "./styled";

const Filters = () => {
  const [fields, setFields] = useState([]);
  const [dataStart, setDataStart] = useState(new Date());
  const [params, setParams] = useState(false);
  const [limitSize, setLimitSize] = useState(false);
  const [offsetIsNegative, setOffsetIsNegative] = useState(false);
  const [generalError, setGeneralError] = useState(false);

  const dispatch = useDispatch();

  const token = useSelector((state) => state.authentication.token);

  useEffect(() => {
    const requestFilters = FiltersApiService();
    const { getFilters } = requestFilters;

    getFilters()
      .then((response) => {
        setFields(response);
      })
      .catch((error) => {
        return error;
      });
  }, []);

  const updateParams = useCallback((id, value) => {
    const filter = {
      [id]: value,
    };

    setParams((prevState) => {
      return { ...prevState, ...filter };
    });
  }, []);

  useEffectUpdate(() => {
    const requestPlaylists = PlaylistsApiService();
    const { getPlaylists } = requestPlaylists;

    getPlaylists(token, params)
      .then((data) => {
        dispatch(updatePlaylists(data));
        dispatch(removeFilteredPlaylists());
      })
      .catch(() => {
        setGeneralError(true);
        dispatch(removePlaylists());
      });
  }, [params]);

  const handleInputChange = useCallback(
    (e, id) => {
      const { value } = e.target;

      if (id === "offset" && Number(value) < 0) {
        return setOffsetIsNegative(true);
      }
      setOffsetIsNegative(false);

      if (id === "limit" && (Number(value) > 50 || Number(value) < 1)) {
        return setLimitSize(true);
      }
      setLimitSize(false);

      if (id === "country" && value === "en_US") {
        return updateParams(id, "US");
      }

      if (value !== "") {
        return updateParams(id, value);
      }
    },
    [updateParams]
  );

  const handleDateChange = useCallback(
    (date, id) => {
      const convertedDate = date.toISOString();

      updateParams(id, convertedDate);

      setDataStart(date);
    },
    [updateParams]
  );

  const fieldsSize = useMemo(() => {
    return fields.length === 0;
  }, [fields]);

  return (
    <S.Filters>
      <S.FiltersBox>
        {fieldsSize && <Loader />}

        {offsetIsNegative && (
          <S.FiltersValidation>
            {filtersContainerData.errors.offset}
          </S.FiltersValidation>
        )}

        {limitSize && (
          <S.FiltersValidation>
            {filtersContainerData.errors.limit}
          </S.FiltersValidation>
        )}

        {generalError && (
          <S.FiltersValidation>
            {filtersContainerData.errors.general}
          </S.FiltersValidation>
        )}

        {fields.map((item) => (
          <S.FiltersItem key={item.id} fieldType={item.id}>
            {getFilterField(
              item,
              (e) => handleInputChange(e, item.id),
              (date) => handleDateChange(date, item.id),
              dataStart
            )}
          </S.FiltersItem>
        ))}
      </S.FiltersBox>
    </S.Filters>
  );
};

export default Filters;
