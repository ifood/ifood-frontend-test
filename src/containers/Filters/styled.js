import styled from "styled-components";
import media from "styled-media-query";

import * as C from "constants/styles/colors";
import * as T from "constants/styles/typography";
import { SPACINGS } from "constants/styles/spacings";

export const Filters = styled.div`
  position: relative;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 324px;
    z-index: -1;
    background: ${C.GREY_LIGHT};
  }
`;

export const FiltersTitle = styled.h2`
  color: ${C.GREY};
  text-align: center;
  padding: ${SPACINGS.xl} 0;
  ${T.TITLE_LARGE};

  ${media.lessThan("large")`
    ${T.TITLE_MEDIUM};
  `}
`;

export const FiltersBox = styled.div`
  box-shadow: 0px 2px 40px rgba(182, 175, 175, 0.35);
  padding: ${SPACINGS.xl};
  max-width: 845px;
  width: 90%;
  margin: 0 auto;
  background: ${C.LIGHT};
  display: flex;
  flex-wrap: wrap;
  border-radius: 5px;

  ${media.lessThan("large")`
    padding: ${SPACINGS.lg};
  `}
`;

export const FiltersItem = styled.div`
  width: 100%;

  :not(:last-child) {
    width: calc(50% - 14px);
    margin: 0 auto 28px;
  }

  :nth-child(even) {
    margin-left: 28px;
  }

  ${media.lessThan("large")`
    width: 100%
  `}
`;
