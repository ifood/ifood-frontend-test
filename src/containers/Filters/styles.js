import styled from 'styled-components';

import { colors, spacing, borderRadius } from '../../styles/theme';

export const Wrapper = styled.div`
  background-color: ${colors.base.white};
  padding: ${spacing.s3};
  border-radius: ${borderRadius};
  box-shadow: 0 0 15px 1px ${colors.muted.m1};
`;
