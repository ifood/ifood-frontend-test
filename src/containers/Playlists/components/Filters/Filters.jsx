
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DatePicker } from 'antd';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import SelectField from './SelectField';
import 'antd/lib/date-picker/style/css';


const Container = styled.div`
  flex-grow: 1;
  max-width: 960px;
  margin: 0 auto;
  margin-top: 50px;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  padding: 24px;
`;

const DatePickerContainer = styled.div`
  margin-top: 24px;
`;

const FILTER_FIELDS = {
  LOCALE: 'locale',
  COUNTRY: 'country',
  LIMIT: 'limit',
  OFFSET: 'offset',
  TIMESTAMP: 'timestamp',
};

class Filters extends Component {
  state = {
    searchTerm: '',
    locale: '',
    country: '',
    limit: '',
    offset: '',
    timestamp: '',
  }

  handleChange = name => (event) => {
    const { onChange } = this.props;
    this.setState({
      [name]: event.target.value,
    });

    onChange({
      ...this.state,
      [name]: event.target.value,
    });
  }

  handleDateChange = (value, dateString) => {
    const { onChange } = this.props;
    this.setState({
      [FILTER_FIELDS.TIMESTAMP]: dateString,
    });

    onChange({
      ...this.state,
      [FILTER_FIELDS.TIMESTAMP]: dateString.replace(' ', 'T'),
    });
  }

  onSearchChange = (event) => {
    const { onSearch } = this.props;
    this.setState({ searchTerm: event.target.value });

    onSearch(event.target.value);
  }

  renderFields = () => {
    const { metaFilters } = this.props;
    const {
      locale,
      country,
      limit,
      offset,
    } = this.state;

    if (!metaFilters) {
      return null;
    }

    return metaFilters.map((field) => {
      let filterField = null;
      switch (field.id) {
        case FILTER_FIELDS.LOCALE:
          filterField = (
            <Grid item xs={12} sm={6} key={field.id}>
              <SelectField
                onChange={this.handleChange(FILTER_FIELDS.LOCALE)}
                value={locale}
                field={field}
              />
            </Grid>
          );
          break;
        case FILTER_FIELDS.COUNTRY:
          filterField = (
            <Grid item xs={12} sm={6} key={field.id}>
              <SelectField
                onChange={this.handleChange(FILTER_FIELDS.COUNTRY)}
                value={country}
                field={field}
              />
            </Grid>
          );
          break;
        case FILTER_FIELDS.LIMIT:
          filterField = (
            <Grid item xs={12} sm={4} key={field.id}>
              <TextField
                id={field.id}
                label={field.name}
                onChange={this.handleChange(FILTER_FIELDS.LIMIT)}
                value={limit}
                type="number"
                inputProps={{
                  min: field.validation.min,
                  max: field.validation.max,
                }}
                fullWidth
                margin="normal"
              />
            </Grid>
          );
          break;
        case FILTER_FIELDS.OFFSET:
          filterField = (
            <Grid item xs={12} sm={4} key={field.id}>
              <TextField
                id={field.id}
                label={field.name}
                onChange={this.handleChange(FILTER_FIELDS.OFFSET)}
                value={offset}
                type="number"
                inputProps={{
                  min: 0,
                }}
                fullWidth
                margin="normal"
              />
            </Grid>
          );
          break;
        case FILTER_FIELDS.TIMESTAMP:
          filterField = (
            <Grid item xs={12} sm={4} key={field.id}>
              <DatePickerContainer>
                <DatePicker
                  showTime
                  format="YYYY-MM-DD HH:mm:ss"
                  placeholder="Select Time"
                  onChange={this.handleDateChange}
                />
              </DatePickerContainer>
            </Grid>
          );
          break;
        default:
          break;
      }
      return filterField;
    });
  }

  render() {
    const { searchTerm } = this.state;
    return (
      <Container>
        <Grid
          container
          direction="row"
          justify="center"
          alignItems="flex-start"
          spacing={24}
        >
          <Grid item xs={12}>
            <TextField
              id="searchTerm"
              label="Search"
              type="search"
              value={searchTerm}
              onChange={this.onSearchChange}
              fullWidth
              margin="normal"
            />
          </Grid>
          {this.renderFields()}
        </Grid>
      </Container>
    );
  }
}

Filters.propTypes = {
  metaFilters: PropTypes.array,
  onChange: PropTypes.func,
  onSearch: PropTypes.func,
};

Filters.defaultProps = {
  metaFilters: null,
  onChange: () => false,
  onSearch: () => false,
};

export default Filters;
