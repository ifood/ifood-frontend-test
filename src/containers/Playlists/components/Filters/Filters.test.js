import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Filters from './Filters';

describe('<Filters />', () => {
  const mocks = {};

  beforeEach(() => {
    mocks.props = {
      onChange: jest.fn(),
      onSearch: jest.fn(),
      metaFilters: [{
        id: 'locale',
        name: 'Locale',
        values: [{
          value: 'en_AU',
          name: 'en_AU',
        },
        {
          value: 'de_DE',
          name: 'de_DE ',
        },
        {
          value: 'pt_BR',
          name: 'pt_BR',
        },
        {
          value: 'fr_FR',
          name: 'fr_FR',
        },
        {
          value: 'en_US',
          name: 'en_US',
        },
        {
          value: 'es_AR',
          name: 'es_AR',
        }],
      },
      {
        id: 'country',
        name: 'País',
        values: [
          {
            value: 'AU',
            name: 'Australia',
          },
          {
            value: 'DE',
            name: 'Alemanhã',
          },
          {
            value: 'BR',
            name: 'Brasil',
          },
          {
            value: 'PT',
            name: 'Portugal',
          },
          {
            value: 'en_US',
            name: 'EUA',
          },
          {
            value: 'RU',
            name: 'Rússia',
          },
        ],
      },
      {
        id: 'timestamp',
        name: 'Data e Horário',
        validation: {
          primitiveType: 'STRING',
          entityType: 'DATE_TIME',
          pattern: 'yyyy-MM-ddTHH:mm:ss',
        },
      },
      {
        id: 'limit',
        name: 'Quantidade',
        validation: {
          primitiveType: 'INTEGER',
          min: 1,
          max: 50,
        },
      },
      {
        id: 'offset',
        name: 'Página',
        validation: {
          primitiveType: 'INTEGER',
        },
      }],
    };
  });

  describe('render()', () => {
    it('renders the content correctly', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should call renderFields with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      wrapper.instance().renderFields = jest.fn();
      wrapper.setProps({ metaFilters: mocks.props.metaFilters });
      expect(wrapper.instance().renderFields).toHaveBeenCalledWith();
    });
  });

  describe('handleChange()', () => {
    it('should call setState with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      const event = {
        target: {
          value: 'pt_BR',
        },
      };
      wrapper.instance().setState = jest.fn();
      wrapper.instance().handleChange('locale')(event);
      expect(wrapper.instance().setState).toHaveBeenCalledWith({ locale: event.target.value });
    });

    it('should call onChange with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      const event = {
        target: {
          value: 'pt_BR',
        },
      };

      wrapper.instance().handleChange('locale')(event);
      expect(mocks.props.onChange).toHaveBeenCalledWith({
        ...wrapper.state(),
        locale: event.target.value,
      });
    });
  });

  describe('handleDateChange()', () => {
    it('should call setState with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      wrapper.instance().setState = jest.fn();
      const dateString = '2018-10-28 10:25:12';
      wrapper.instance().handleDateChange(undefined, dateString);
      expect(wrapper.instance().setState).toHaveBeenCalledWith({ timestamp: dateString });
    });

    it('should call onChange with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      const dateString = '2018-10-28 10:25:12';
      wrapper.instance().handleDateChange(undefined, dateString);
      expect(mocks.props.onChange).toHaveBeenCalledWith({
        ...wrapper.state(),
        timestamp: dateString.replace(' ', 'T'),
      });
    });
  });

  describe('onSearchChange()', () => {
    it('should call setState with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      const event = {
        target: {
          value: 'term',
        },
      };
      wrapper.instance().setState = jest.fn();
      wrapper.instance().onSearchChange(event);
      expect(wrapper.instance().setState).toHaveBeenCalledWith({ searchTerm: event.target.value });
    });

    it('should call onSearch with expected params', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      const event = {
        target: {
          value: 'pt_BR',
        },
      };
      wrapper.instance().onSearchChange(event);
      expect(mocks.props.onSearch).toHaveBeenCalledWith(event.target.value);
    });
  });

  describe('renderFields()', () => {
    it('should return the content as expected according to the meta filters', () => {
      const wrapper = shallow(<Filters {...mocks.props} />);
      wrapper.setProps({ metaFilters: mocks.props.metaFilters });
      const content = wrapper.instance().renderFields();
      expect(content).toMatchSnapshot();
    });
  });
});
