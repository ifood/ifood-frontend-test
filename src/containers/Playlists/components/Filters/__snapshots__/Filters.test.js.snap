// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> render() renders the content correctly 1`] = `
<styled.div>
  <WithStyles(Grid)
    alignItems="flex-start"
    container={true}
    direction="row"
    justify="center"
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <TextField
        fullWidth={true}
        id="searchTerm"
        label="Search"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        type="search"
        value=""
        variant="standard"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="locale"
      sm={6}
      xs={12}
    >
      <SelectField
        field={
          Object {
            "id": "locale",
            "name": "Locale",
            "values": Array [
              Object {
                "name": "en_AU",
                "value": "en_AU",
              },
              Object {
                "name": "de_DE ",
                "value": "de_DE",
              },
              Object {
                "name": "pt_BR",
                "value": "pt_BR",
              },
              Object {
                "name": "fr_FR",
                "value": "fr_FR",
              },
              Object {
                "name": "en_US",
                "value": "en_US",
              },
              Object {
                "name": "es_AR",
                "value": "es_AR",
              },
            ],
          }
        }
        onChange={[Function]}
        value=""
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="country"
      sm={6}
      xs={12}
    >
      <SelectField
        field={
          Object {
            "id": "country",
            "name": "País",
            "values": Array [
              Object {
                "name": "Australia",
                "value": "AU",
              },
              Object {
                "name": "Alemanhã",
                "value": "DE",
              },
              Object {
                "name": "Brasil",
                "value": "BR",
              },
              Object {
                "name": "Portugal",
                "value": "PT",
              },
              Object {
                "name": "EUA",
                "value": "en_US",
              },
              Object {
                "name": "Rússia",
                "value": "RU",
              },
            ],
          }
        }
        onChange={[Function]}
        value=""
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="timestamp"
      sm={4}
      xs={12}
    >
      <styled.div>
        <PickerWrapper
          format="YYYY-MM-DD HH:mm:ss"
          inputPrefixCls="ant-input"
          locale={Object {}}
          onChange={[Function]}
          onOk={[Function]}
          onOpenChange={[Function]}
          placeholder="Select Time"
          popupStyle={Object {}}
          prefixCls="ant-calendar"
          showTime={true}
          transitionName="slide-up"
        />
      </styled.div>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="limit"
      sm={4}
      xs={12}
    >
      <TextField
        fullWidth={true}
        id="limit"
        inputProps={
          Object {
            "max": 50,
            "min": 1,
          }
        }
        label="Quantidade"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        type="number"
        value=""
        variant="standard"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      key="offset"
      sm={4}
      xs={12}
    >
      <TextField
        fullWidth={true}
        id="offset"
        inputProps={
          Object {
            "min": 0,
          }
        }
        label="Página"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        type="number"
        value=""
        variant="standard"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</styled.div>
`;

exports[`<Filters /> renderFields() should return the content as expected according to the meta filters 1`] = `
Array [
  <WithStyles(Grid)
    item={true}
    sm={6}
    xs={12}
  >
    <SelectField
      field={
        Object {
          "id": "locale",
          "name": "Locale",
          "values": Array [
            Object {
              "name": "en_AU",
              "value": "en_AU",
            },
            Object {
              "name": "de_DE ",
              "value": "de_DE",
            },
            Object {
              "name": "pt_BR",
              "value": "pt_BR",
            },
            Object {
              "name": "fr_FR",
              "value": "fr_FR",
            },
            Object {
              "name": "en_US",
              "value": "en_US",
            },
            Object {
              "name": "es_AR",
              "value": "es_AR",
            },
          ],
        }
      }
      onChange={[Function]}
      value=""
    />
  </WithStyles(Grid)>,
  <WithStyles(Grid)
    item={true}
    sm={6}
    xs={12}
  >
    <SelectField
      field={
        Object {
          "id": "country",
          "name": "País",
          "values": Array [
            Object {
              "name": "Australia",
              "value": "AU",
            },
            Object {
              "name": "Alemanhã",
              "value": "DE",
            },
            Object {
              "name": "Brasil",
              "value": "BR",
            },
            Object {
              "name": "Portugal",
              "value": "PT",
            },
            Object {
              "name": "EUA",
              "value": "en_US",
            },
            Object {
              "name": "Rússia",
              "value": "RU",
            },
          ],
        }
      }
      onChange={[Function]}
      value=""
    />
  </WithStyles(Grid)>,
  <WithStyles(Grid)
    item={true}
    sm={4}
    xs={12}
  >
    <ForwardRef>
      <PickerWrapper
        format="YYYY-MM-DD HH:mm:ss"
        inputPrefixCls="ant-input"
        locale={Object {}}
        onChange={[Function]}
        onOk={[Function]}
        onOpenChange={[Function]}
        placeholder="Select Time"
        popupStyle={Object {}}
        prefixCls="ant-calendar"
        showTime={true}
        transitionName="slide-up"
      />
    </ForwardRef>
  </WithStyles(Grid)>,
  <WithStyles(Grid)
    item={true}
    sm={4}
    xs={12}
  >
    <TextField
      fullWidth={true}
      id="limit"
      inputProps={
        Object {
          "max": 50,
          "min": 1,
        }
      }
      label="Quantidade"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="number"
      value=""
      variant="standard"
    />
  </WithStyles(Grid)>,
  <WithStyles(Grid)
    item={true}
    sm={4}
    xs={12}
  >
    <TextField
      fullWidth={true}
      id="offset"
      inputProps={
        Object {
          "min": 0,
        }
      }
      label="Página"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      type="number"
      value=""
      variant="standard"
    />
  </WithStyles(Grid)>,
]
`;
